### DISCUSSION TITLE
Feature request: allow extending `defaultEventNames` mapping

### DISCUSSION THREAD

**Discussion by grncdr** [stimulusjs/stimulus#818] [General]
The logic for shorthand/default event names in stimulus is currently defined here: https://github.com/hotwired/stimulus/blob/f7bfc358e5df2cd4623ae47325254e4076d61148/src/core/action.ts#L89-L97

This is pretty handy, but if you are using a library of custom elements (e.g. shoelace) you have to turn `<button data-action="controller#method"></button>` into `<sl-button data-action="click->controller#method"></sl-button>`.

It's a tiny thing, but if stimulus were to simply export that mapping it would be possible for users to do something like:

```js
import { defaultEventNames } from '@hotwired/stimulus'

defaultEventNames['sl-button'] = () => 'click'
```

```html
<sl-button data-action="controller#method"></sl-button>
```

If exporting the mapping directly is exposing too many implementation details, a function like `setDefaultEventName(elementName, stringOrCallback)` would also be pretty nice.

If that makes sense to the maintainers I can take a crack at a PR.

**Comment by marcoroth** [comment]
Thanks for opening this issue @grncdr!

I wouldn't be opposed to add support for something like this. I think it would make sense to provide some sort of public interface instead of just exporting the object.
